src/main/java/com/wildfire/main/WildfireHelper.java

```diff
@@ -33,16 +33,45 @@
 import net.minecraft.component.type.AttributeModifiersComponent;
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.TriState;
+import net.minecraft.util.math.Direction;
 import net.minecraft.util.math.MathHelper;
 
+import java.util.Arrays;
 import java.util.concurrent.ThreadLocalRandom;
 import java.util.function.Function;
 
 public final class WildfireHelper {
+
     private WildfireHelper() {
         throw new UnsupportedOperationException();
     }
 
+    public static final Direction[] SERIALIZED_DIRECTIONS = {
+            Direction.EAST,
+            Direction.WEST,
+            Direction.DOWN,
+            Direction.UP,
+            Direction.NORTH
+    };
+
+    public static int getDirectionIndex(Direction dir) {
+        for (int i = 0; i < SERIALIZED_DIRECTIONS.length; i++) {
+            if (SERIALIZED_DIRECTIONS[i] == dir) {
+                return i;
+            }
+        }
+        return -1; // not found
```

Comment:
could be simplified (or arguably just entirely removed) if this array is turned into a `List`
```suggestion
        return SERIALIZED_DIRECTIONS.indexOf(dir);
```